src/device-registry/utils/device.util.js

```
@@ -459,10 +459,37 @@ const createDevice = {
       const { tenant, path, limit, skip } = request.query;
-      const filter = generateFilter.devices(request, next);
+
+      // Get the base filter from your existing function
+      const baseFilter = generateFilter.devices(request, next);
+
+      // Merge with resource access filter if available
+      let filter = { ...baseFilter };
+
+      // If auth middleware has added resource filters, apply them
+      if (request.resourceFilters && request.resourceFilters.devices) {
+        // Merge the ID filters - if both filters have _id conditions, use $and to combine them
```

Biome (1.9.4)

[error] Change to an optional chain.

Unsafe fix: Change to an optional chain.


(lint/complexity/useOptionalChain)